src/compiler/JSCodeGenVisitor.js

```diff
@@ -25,7 +27,10 @@ function escapeJavaString(s) {
 module.exports = class JSCodeGenVitor {
   constructor() {
     this._result = '';
+    this._templates = '';
     this._indentLevel = 0;
+    this._lastIndentLevel = 0;
+    this._inFunctionBlock = false;
```

Comment:
`_inFunctionBlock` determines whether each call to the `visit()` method should write its JavaScript to the `result` stream or the new `template` stream.